sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/InputFileFilter.java

```diff
@@ -28,6 +28,7 @@
  */
 @ScannerSide
 @ExtensionPoint
+@FunctionalInterface
 public interface InputFileFilter {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Drop this interface in favor of "java.util.function.Predicate<InputFile>". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1711)